File: examples/node/webhook-retry.mjs
import { ChainFX } from "../../sdk/node/index.js";

const chainfx = new ChainFX({
  apiKey: process.env.CHAINFX_API_KEY || "sk_test_chainfx_local",
  baseUrl: process.env.CHAINFX_API_BASE_URL || "http://localhost:8080"
});

const result = await chainfx.retryWebhook({
  orderId: process.env.CHAINFX_ORDER_ID,
  side: process.env.CHAINFX_ORDER_SIDE || "buy",
  event: "payment.completed",
  targetUrl: process.env.CHAINFX_WEBHOOK_TARGET_URL
});

console.log(result);

File: examples/python/webhook_retry.py
import os
import sys

sys.path.append(os.path.join(os.path.dirname(__file__), "..", "..", "sdk", "python"))

from chainfx import ChainFX


chainfx = ChainFX(
    api_key=os.getenv("CHAINFX_API_KEY", "sk_test_chainfx_local"),
    base_url=os.getenv("CHAINFX_API_BASE_URL", "http://localhost:8080"),
)

result = chainfx.retry_webhook(
    order_id=os.environ["CHAINFX_ORDER_ID"],
    side=os.getenv("CHAINFX_ORDER_SIDE", "buy"),
    event="payment.completed",
    target_url=os.getenv("CHAINFX_WEBHOOK_TARGET_URL"),
)
print(result)
